airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/StreamNamePair.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk
+
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair
+import io.airbyte.protocol.models.v0.StreamDescriptor
+
+/**
+ * [StreamNamePair] is equivalent to [AirbyteStreamNameNamespacePair].
+ *
+ * This exists to avoid coupling the Bulk CDK code too closely to the Airbyte Protocol objects.
+ */
+data class StreamNamePair(val name: String, val namespace: String?) {
+    override fun toString(): String {
+        return if (namespace == null) name else "${namespace}_${name}"
```

Comment:
Couldn't agree more! However, it's how `AirbyteStreamNameNamespacePair` chooses to stringify itself. I don't want to change this here, it could subvert expectations.

Reply:
why do we need to carry over a bad divider, if it's a different object?

Reply:
Because it ends up in logs and possibly other user-visible output.
If you can propose a stringification that's obviously superior then by all means let's disrupt.

Reply:
I think `:` is unusable as a table or schema name. Another very common one is `.`

Reply:
Dots are good in my book. Will do.